From d22603ea858a72719ced5836ba94c8c18243461a Mon Sep 17 00:00:00 2001
From: Md Arif <111168803+sabamdarif@users.noreply.github.com>
Date: Mon, 9 Jun 2025 23:54:44 +0530
Subject: [PATCH] fix_tmp_paths

---
 src/main.rs | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/main.rs b/src/main.rs
index 495e61b4..22f1d5f9 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -409,7 +409,7 @@ fn fetch_url(url: &String, file_name: String) -> Result<()> {
     Ok(())
 }
 fn install_theme(downloaddetail: &DownloadDetail, themetype: &Catalog) -> Result<()> {
-    let mut path = String::from("/tmp/themedownloadfiles/");
+    let mut path = String::from("/data/data/com.termux/files/usr/tmp/themedownloadfiles/");
     path.push_str(themetype.to_string());
     path.push_str("/");
 
@@ -635,7 +635,7 @@ fn downloadthumb(each_product: &Product) -> Result<()> {
         return Ok(());
     }
     let firstimage = firstimage.unwrap();
-    let save_path = "/tmp/themeinstaller/cache/".to_string() + &firstimage;
+    let save_path = "/data/data/com.termux/files/usr/tmp/themeinstaller/cache/".to_string() + &firstimage;
     if !std::path::Path::new(&save_path).exists() {
         let mut save_dir = save_path.to_string();
         save_dir.push_str(&firstimage);
@@ -656,7 +656,7 @@ fn downloadotherimages(each_product: &Product) -> Result<()> {
     //println!("Got inside Download Thumbnail");
 
     for each_image in &each_product.previewpics[1..] {
-        let save_path = "/tmp/themeinstaller/cache/".to_string() + &each_image;
+        let save_path = "/data/data/com.termux/files/usr/tmp/themeinstaller/cache/".to_string() + &each_image;
         if !std::path::Path::new(&save_path).exists() {
             let mut save_dir = save_path.to_string();
             save_dir.push_str(&each_image);
@@ -691,7 +691,7 @@ fn _downloadthumbs(products: Vec<Product>) -> Result<()> {
             //println!("Image link : {:?}", image_small.clone().unwrap());
             //println!("In async tokio");
             for image_small in image_small_list {
-                let save_path = "/tmp/themeinstaller/cache/".to_string() + &image_small;
+                let save_path = "/data/data/com.termux/files/usr/tmp/themeinstaller/cache/".to_string() + &image_small;
                 if !std::path::Path::new(&save_path).exists() {
                     let mut save_dir = save_path.to_string();
                     save_dir.push_str(&image_small);
@@ -813,7 +813,7 @@ fn build_search_page(
 }
 // contentbox function
 fn build_flowbox_for_page(each_product: &Product, flowbox: &FlowBox, window: &ApplicationWindow) {
-    let imgpath = "/tmp/themeinstaller/cache/".to_string() + &each_product.previewpics[0];
+    let imgpath = "/data/data/com.termux/files/usr/tmp/themeinstaller/cache/".to_string() + &each_product.previewpics[0];
     let img = Picture::builder()
         .valign(Align::Center)
         .hexpand_set(false)
@@ -1112,7 +1112,7 @@ fn build_flowbox_for_page(each_product: &Product, flowbox: &FlowBox, window: &Ap
 
         //Insert Images in dialog body
         let total_preview_pics = product.previewpics.len();
-        let imgpath = "/tmp/themeinstaller/cache/".to_string() + &product.previewpics[0];
+        let imgpath = "/data/data/com.termux/files/usr/tmp/themeinstaller/cache/".to_string() + &product.previewpics[0];
         let img = Picture::builder()
             .valign(Align::Center)
             .hexpand_set(true)
@@ -1179,7 +1179,7 @@ fn build_flowbox_for_page(each_product: &Product, flowbox: &FlowBox, window: &Ap
                 *current_index -= 1;
             }
             let current_index = *current_index as usize;
-            let imgpath = "/tmp/themeinstaller/cache/".to_string() + &previewpics[current_index];
+            let imgpath = "/data/data/com.termux/files/usr/tmp/themeinstaller/cache/".to_string() + &previewpics[current_index];
             img_prev.set_filename(Some(&std::path::Path::new(imgpath.as_str())));
         });
 
@@ -1196,7 +1196,7 @@ fn build_flowbox_for_page(each_product: &Product, flowbox: &FlowBox, window: &Ap
             }
             let current_index = *current_index as usize;
             let imgpath =
-                "/tmp/themeinstaller/cache/".to_string() + &previewpics_next[current_index];
+                "/data/data/com.termux/files/usr/tmp/themeinstaller/cache/".to_string() + &previewpics_next[current_index];
             img_next.set_filename(Some(&std::path::Path::new(imgpath.as_str())));
         });
 
-- 
2.49.0

